prover_rust/src/key_signer.rs

```diff
@@ -0,0 +1,108 @@
+use std::path::Path;
+
+use anyhow::Result;
+use ethers_core::k256::{
+    ecdsa::{signature::hazmat::PrehashSigner, RecoveryId, Signature, SigningKey},
+    elliptic_curve::{sec1::ToEncodedPoint, FieldBytes},
+    Secp256k1,
+    PublicKey,
+    SecretKey
+};
+
+use hex::ToHex;
+use tiny_keccak::{Hasher, Keccak};
+use eth_types::{H256, U256};
+
+use ethers_core::types::Signature as EthSignature;
```

Comment:
need to format the code and add some format tools to CI. cc @HAOYUatHZ do you know any useful tools?